app/controllers/users/circuitverse_controller.rb

```diff
@@ -11,6 +11,10 @@ class Users::CircuitverseController < ApplicationController
   def index
     @profile = ProfileDecorator.new(@user)
     @projects = @user.rated_projects
+    public_projects = @user.projects.map do |project|
+      policy(project).check_direct_view_access?
+    end
+    @public_projects_exist = public_projects.include?(true)
```

Comment:
If you want to check if public projects exists, most perfomant way is to use

```suggestion
   public_projects_exist = @user.projects.open.exists?
```

Reply:
Yes Sir this is correct,
But with this change, the admin user will not able to see his own private projects(if he has only private projects) instead it will show the `No result image`.

https://user-images.githubusercontent.com/76155456/146793412-fdbca57f-5a17-4cc9-b76a-687f91210c8f.mov

See, when the User has 1 private and 1 public project, there is no problem,
But deleting the public one, then it will not show the private project to the User.

Please check.

Reply:
That means you were setting wrong value to public_projects_exist, right ?

Reply:
No, I think this is correct.

Reply:
Hey @tachyons,
I have updated this PR,
I was wrong, about the above statement ( > No sir, I think this is correct.)
I am sorry for that 😔, learning from my mistakes
Please review 
Thanks